src/Containers/MovieListPage/MovieListPage.jsx

```diff
@@ -1,77 +1,64 @@
-import { MovieDetails } from '../../Components/MovieDetails/MovieDetails';
 import { MovieTile } from '../../Components/MovieTile/MovieTile';
 import { SortControl } from '../../Components/SortControl/SortControl';
-import { Dialog } from '../../Components/Dialog/Dialog';
-import { MovieForm } from '../../Components/MovieForm/MovieForm';
 import styles from './MovieListPage.module.css';
-import { useFetchMovies } from '../../customHooks/movies';
+import { useFetchMovies } from '../../customHooks/moviesHooks';
+import { Outlet, useSearchParams } from 'react-router-dom';
 
-export function MovieListPage() {
-  const [selectedGenre, setSelectedGenre] = useState(genres[0]);
-  const [searchQuery, setSearchQuery] = useState('');
-  const [selectedSortOption, setSelectedSortOption] = useState(sortOptions[0].value);
-  const [selectedMovie, setSelectedMovie] = useState();
+const defaultQuery = {
+  search: '',
+  genre: genres[0],
+  sortBy: sortOptions[0].value,
+};
 
-  const [isAddMovieDialogOpen, setIsAddMovieDialogOpen] = useState(false);
-  const movies = useFetchMovies(searchQuery, selectedSortOption, selectedGenre);
+export function MovieListPage() {
+  const [searchParams, setSearchParams] = useSearchParams(defaultQuery);
+  const movies = useFetchMovies(searchParams.get('search'), searchParams.get('sortBy'), searchParams.get('genre'));
 
   return (
-    <div>
+    <div className={styles.container}>
       <div className={styles.headerTile}>
-        {selectedMovie ? (
-          <>
-            <MovieDetails
-              imageUrl={selectedMovie.imageUrl}
-              movieName={selectedMovie.movieName}
-              releaseYear={selectedMovie.releaseYear}
-              rating={selectedMovie.rating}
-              duration={selectedMovie.duration}
-              description={selectedMovie.description}
-            />
-            <button className={styles.headerButton} onClick={() => setSelectedMovie(null)}>
-              Search
-            </button>
-          </>
-        ) : (
-          <>
-            <SearchForm initialSearchQuery={searchQuery} onSearch={(value) => setSearchQuery(value)} />
-            <button className={styles.headerButton} onClick={() => setIsAddMovieDialogOpen(true)}>
-              Add movie
-            </button>
-          </>
-        )}
+        <Outlet />
       </div>
 
       <div className={styles.movieListTile}>
         <div className={styles.filters}>
-          <GenreSelect allGenres={genres} selectedGenre={selectedGenre} onSelect={(genre) => setSelectedGenre(genre)} />
-          <SortControl selectedOption={selectedSortOption} onSelect={(value) => setSelectedSortOption(value)} />
+          <GenreSelect
+            allGenres={genres}
+            selectedGenre={searchParams.get('genre')}
+            onSelect={(genre) =>
+              setSearchParams((params) => {
+                params.set('genre', genre);
+                return params;
+              })
+            }
```

Comment:
All these values can be used directly in `GenreSelect` without passing them as props.
Now there is no need in splitting this logic between 2 components. It can be kept in one place - `GenreSelect`.
The state of `selectedGenre` is global now. Props drilling can be avoided.